### ISSUE TITLE
Examples for ETag matching (Trac #71)

### ISSUE THREAD

**Issue by mnot** [httpwg/httpbis-issues#71]
People seem to be confused about the weak matching function; we should have examples to illustrate it.

Migrated from https://trac.ietf.org/ticket/71
```json
{
    "status": "closed",
    "changetime": "2009-07-17T15:50:58",
    "_ts": "1247845858000000",
    "description": "People seem to be confused about the weak matching function; we should have examples to illustrate it.",
    "reporter": "mnot@pobox.com",
    "cc": "",
    "resolution": "fixed",
    "time": "2007-12-22T02:59:36",
    "component": "p4-conditional",
    "summary": "Examples for ETag matching",
    "priority": "",
    "keywords": "",
    "milestone": "04",
    "owner": "julian.reschke@gmx.de",
    "type": "design",
    "severity": ""
}
```

**Comment by mnot** [comment]
Trac update at 20071222T03:00:05: mnot@pobox.com commented:

> 
> ```text
> +--------+--------+-------------------+-----------------+
> | ETag 1 | ETag 2 | Strong Comparison | Weak Comparison |
> +--------+--------+-------------------+-----------------+
> | W/"1"  | W/"1"  | no match          | match           |
> |        |        |                   |                 |
> | W/"1"  | W/"2"  | no match          | no match        |
> |        |        |                   |                 |
> | W/"1"  | "1"    | no match          | match           |
> |        |        |                   |                 |
> | "1"    | "1"    | match             | match           |
> +--------+--------+-------------------+-----------------+
> ```

**Comment by mnot** [comment]
Trac update at 20080104T06:05:05: 
* mnot@pobox.com changed component from "" to "conditional"
* mnot@pobox.com changed milestone from "" to "unassigned"
* mnot@pobox.com changed owner from "" to ""

**Comment by mnot** [comment]
Trac update at 20080205T03:06:05: fielding@gbiv.com changed type from "design" to "editorial"

**Comment by mnot** [comment]
Trac update at 20080728T13:34:59: 
* julian.reschke@gmx.de commented:

> Proposal:
> 
> 1) Replace
> 
> "    * The strong comparison function: in order to be considered equal, both validators MUST be identical in every way, and both MUST NOT be weak.
>     * The weak comparison function: in order to be considered equal, both validators MUST be identical in every way, but either or both of them MAY be tagged as "weak" without affecting the result."
> 
> and
> 
> "    * The strong comparison function: in order to be considered equal, both opaque-tags MUST be identical character-by-character, and both MUST NOT be weak.
>     * The weak comparison function: in order to be considered equal, both opaque-tags MUST be identical character-by-character."
> 
> This uses "opaque-token" instead of "validator" in the definition of the matching functions.
> 
> 
> 2) Add the table below.
> 


* julian.reschke@gmx.de changed owner from "" to "julian.reschke@gmx.de"

**Comment by mnot** [comment]
Trac update at 20080728T13:36:04: 
* julian.reschke@gmx.de commented:

> Proposal:
> 
> 1) Replace
> 
> " * The strong comparison function: in order to be considered equal, both validators MUST be identical in every way, and both MUST NOT be weak.
> 
>     * The weak comparison function: in order to be considered equal, both validators MUST be identical in every way, but either or both of them MAY be tagged as "weak" without affecting the result." 
> 
> by
> 
> " * The strong comparison function: in order to be considered equal, both opaque-tags MUST be identical character-by-character, and both MUST NOT be weak.
> 
>     * The weak comparison function: in order to be considered equal, both opaque-tags MUST be identical character-by-character." 
> 
> This uses "opaque-token" instead of "validator" in the definition of the matching functions.
> 
> 2) Add the table below.
> 


* julian.reschke@gmx.de changed milestone from "unassigned" to "04"
* julian.reschke@gmx.de changed type from "editorial" to "design"

**Comment by mnot** [comment]
Trac update at 20080729T16:26:55: 
* julian.reschke@gmx.de changed attachment from "" to "i71.diff"
* julian.reschke@gmx.de commented:

> proposed change for part 3.

**Comment by mnot** [comment]
Trac update at 20080731T13:37:35: 
* julian.reschke@gmx.de commented:

> Fixed in [298]:
> 
> Resolve #71: Clarify matching functions and add examples (closes #71).


* julian.reschke@gmx.de changed resolution from "" to "fixed"
* julian.reschke@gmx.de changed status from "new" to "closed"

**Comment by mnot** [comment]
Trac update at 20090717T15:50:58: julian.reschke@gmx.de commented:

> From [610]:
> 
> re-add text about how to do the weak match (while resolving #71 it got a bit terse) (related to #71)
>
